### PR TITLE
Movesearch: Add ability to search for moves in more than one Pokemon's learnset

### PR DESCRIPTION
**Pull request by jmclemo6**

### CODE REVIEW

**Comment by jmclemo6** [comment]
I saw some use for this for some OMs but didn't think about how this would affect the param hunts in Scavengers, so this change is awaiting approval by room staff there. Please don't merge until then. :)

**Comment by Zarel** [comment]
This pull request **introduces 3 alerts** when merging 81b7df0a16656a6c6cf2dce226cf32379cef0782 into 233ffd2cba79ac0accc04f33089526b405389cb0 - [view on LGTM.com](https://lgtm.com/projects/g/Zarel/Pokemon-Showdown/rev/pr-88bbd27f7282885efa7c25eb09ff9cdf51a0e3a7)

**new alerts:**

* 3 for Useless conditional

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by jmclemo6** [comment]
The opinion was pretty split. @Zarel you can merge if you want. Really up to you on whether or not a split opinion is good enough to push this through.

**Comment by Zarel** [comment]
I mean, this doesn't look like bad code... I really don't want to make calls on movesearch, though.

@bumbadadabum or @HoeenCoder, opinions?

**Comment by HoeenCoder** [comment]
I'll probably get to this on tuesday (or sooner if i get some time).

File: server/chat-plugins/datasearch.js [hunk 1]
```diff
@@ -882,15 +882,17 @@ function runMovesearch(target, cmd, canAll, message) {
 
 			let template = Dex.getTemplate(target);
 			if (template.exists) {
-				if (Object.keys(lsetData).length) return {reply: "A search can only include one Pok\u00e9mon learnset."};
+				if (targetMons.includes(template.name)) return {reply: "A search cannot include the same Pok\u00e9mon twice."};
 				if (parameters.length > 1) return {reply: "A Pok\u00e9mon learnset cannot have alternative parameters."};
 				if (!template.learnset) template = Dex.getTemplate(template.baseSpecies);
-				lsetData = Object.assign({}, template.learnset);
-				targetMon = template.name;
+				lsetData[template.name] = template.learnset;
```
Comment (HoeenCoder) [inline server/chat-plugins/datasearch.js]:
```suggestion
				lsetData[template.name] = Object.assign({}, template.learnset);
```

**Pull request merged**